pulumi-eks-go/main.go

```diff
@@ -0,0 +1,439 @@
+			SubnetId:     privSubnet1.ID(),
+			RouteTableId: privateRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-priv-2", &ec2.RouteTableAssociationArgs{
+			SubnetId:     privSubnet2.ID(),
+			RouteTableId: privateRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-priv-3", &ec2.RouteTableAssociationArgs{
+			SubnetId:     privSubnet3.ID(),
+			RouteTableId: privateRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Associate Public Subs with Public Route Tables
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-pub-1", &ec2.RouteTableAssociationArgs{
+			SubnetId:     pubSubnet1.ID(),
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-pub-2", &ec2.RouteTableAssociationArgs{
+			SubnetId:     pubSubnet2.ID(),
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-pub-3", &ec2.RouteTableAssociationArgs{
+			SubnetId:     pubSubnet3.ID(),
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Resource: IAM Role
+		// Purpose: An IAM role is an IAM identity that you can create in your account that has specific permissions.
+		// Docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+
+		// IAM Role for EKS
+		eksRole, err := iam.NewRole(ctx, prefix+"eks-iam-eksRole", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
+		    "Version": "2008-10-17",
+		    "Statement": [{
+		        "Sid": "",
+		        "Effect": "Allow",
+		        "Principal": {
+		            "Service": "eks.amazonaws.com"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+		})
+		if err != nil {
+			return err
+		}
+		eksPolicies := []string{
+			"arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+			"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+		}
+		for i, eksPolicy := range eksPolicies {
+			_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf(prefix+"rpa-%d", i), &iam.RolePolicyAttachmentArgs{
+				PolicyArn: pulumi.String(eksPolicy),
+				Role:      eksRole.Name,
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		// Resource: Security Groups
+		// Purpose: A security group acts as a virtual firewall, controlling the traffic that is allowed to reach and leave the resources that it is associated with.
+		// Docs: https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html
+
+		// Create a Security Group that we can use to actually connect to our cluster
+		clusterSg, err := ec2.NewSecurityGroup(ctx, prefix+"-cluster-sg", &ec2.SecurityGroupArgs{
+			VpcId: vpc.ID(),
+			Egress: ec2.SecurityGroupEgressArray{
+				ec2.SecurityGroupEgressArgs{
```

Comment:
Can't remember if this is required, as my route table has source `0.0.0.0/0` & destination as the IGW for pub subnets & NAT GW for priv subnets